### ISSUE TITLE
Ddividing by zero

### ISSUE THREAD

**Issue by z1mEk** [AlbertHakvoort/StorkAir-Zehnder-WHR-930-Domoticz-MQTT#1]
While the script is running is dividing by zero.

Nov 14 13:43:45 pi python3[23822]: ************************
Nov 14 13:43:45 pi python3[23822]: * WHR930 MQTT Domoticz *
Nov 14 13:43:45 pi python3[23822]: ************************
Nov 14 13:43:45 pi python3[23822]: 14-11-2019 12:43:17 INFO: Successfull subscribed to the MQTT topics
Nov 14 13:43:45 pi python3[23822]: Traceback (most recent call last):
Nov 14 13:43:45 pi python3[23822]:   File "/usr/local/bin/whr930.py", line 284, in <module>
Nov 14 13:43:45 pi python3[23822]:     get_fan_status()
Nov 14 13:43:45 pi python3[23822]:   File "/usr/local/bin/whr930.py", line 198, in get_fan_status
Nov 14 13:43:45 pi python3[23822]:     IntakeFanRPM = int(1875000 / int(''.join([str(int(data[9], 16)), str(int(data[10], 16))])))
Nov 14 13:43:45 pi python3[23822]: ZeroDivisionError: division by zero
Nov 14 13:43:46 pi systemd[1]: whr930.service: Main process exited, code=exited, status=1/FAILURE
Nov 14 13:43:46 pi systemd[1]: whr930.service: Unit entered failed state.
Nov 14 13:43:46 pi systemd[1]: whr930.service: Failed with result 'exit-code'.
Nov 14 13:43:56 pi systemd[1]: whr930.service: Service hold-off time over, scheduling restart.
Nov 14 13:43:56 pi systemd[1]: Stopped WHR930.
Nov 14 13:43:56 pi systemd[1]: Started WHR930.

**Comment by anixi** [comment]
Hi , did you solve this issue in the meanwhile. I have the same issue:
Best regards

**Comment by tice17** [comment]
I copied the code and adapted it for my own purposes. I tackled the division by zero error. I think it was caused by the fact that the original script does not take into account that the WHR920/930 is very "talkative", whereas the original script wrongly assumes that the WHR only talks back when spoken to. Not so when your WHR has the "DeLuxe" display. https://gitlab.com/matthijs_rademakers/StorkAir-Zehnder-WHR-930-Domoticz-MQTT/
